bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -68,7 +74,10 @@ public StateDescription getStateDescription(String itemName, Locale locale) {
             if (channel != null) {
                 ChannelType channelType = thingTypeRegistry.getChannelType(channel, locale);
                 if (channelType != null) {
-                    StateDescription stateDescription = channelType.getState();
+                    StateDescription stateDescription = getDynamicStateDescription(channel, locale);
```

Comment:
In the way you implemented it here it will only work if there is a channelType defined. Keep in mind: there are bindings which work perfectly well without channel types. With this PR they actually should get the chance to also provide state descriptions for channels now, so this will be a big improvement. You will just have to move some of the logic out of the `if (channelType != null) {` block.

Reply:
@sjka: I reshuffled the logic. An idea came to my mind in the meantime, that maybe we could pass the original state description as an argument to this provider, so it can return it with modifications of only some parameters. I don't have a use case, but it would be more broad and also work for original case. What do you think?

Reply:
Excellent idea! 
I already had some doubts whether we maybe should try merging both results, but that all felt a bit awkward. Your suggestion of handing over the "static" description (if it exists) makes much more sense, because then the dynamic provide can decide to accept all the attributes it is not interested in.